trino-aws-proxy/src/test/java/io/trino/aws/proxy/server/TestAuthzBeforeRewrite.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.aws.proxy.server;
+
+import com.google.inject.Inject;
+import com.google.inject.Scopes;
+import io.airlift.http.server.testing.TestingHttpServer;
+import io.trino.aws.proxy.server.testing.RequestRewriteUtil;
+import io.trino.aws.proxy.server.testing.TestingS3RequestRewriter;
+import io.trino.aws.proxy.server.testing.TestingTrinoAwsProxyServer;
+import io.trino.aws.proxy.server.testing.containers.S3Container.ForS3Container;
+import io.trino.aws.proxy.server.testing.harness.BuilderFilter;
+import io.trino.aws.proxy.server.testing.harness.TrinoAwsProxyTest;
+import io.trino.aws.proxy.server.testing.harness.TrinoAwsProxyTestCommonModules.WithConfiguredBuckets;
+import io.trino.aws.proxy.spi.security.S3SecurityFacadeProvider;
+import io.trino.aws.proxy.spi.security.SecurityResponse;
+import org.assertj.core.api.InstanceOfAssertFactories;
+import org.junit.jupiter.api.Test;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+import software.amazon.awssdk.services.s3.model.S3Exception;
+
+import java.net.URI;
+import java.util.Optional;
+
+import static com.google.inject.multibindings.OptionalBinder.newOptionalBinder;
+import static io.trino.aws.proxy.server.testing.RequestRewriteUtil.CREDENTIAL_TO_REDIRECT;
+import static io.trino.aws.proxy.server.testing.RequestRewriteUtil.TEST_CREDENTIAL_REDIRECT_BUCKET;
+import static io.trino.aws.proxy.server.testing.RequestRewriteUtil.TEST_CREDENTIAL_REDIRECT_KEY;
+import static io.trino.aws.proxy.server.testing.TestingUtil.TEST_FILE;
+import static io.trino.aws.proxy.server.testing.TestingUtil.assertFileNotInS3;
+import static io.trino.aws.proxy.server.testing.TestingUtil.clientBuilder;
+import static io.trino.aws.proxy.spi.plugin.TrinoAwsProxyServerBinding.s3RequestRewriterModule;
+import static java.util.Objects.requireNonNull;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@TrinoAwsProxyTest(filters = {WithConfiguredBuckets.class, TestAuthzBeforeRewrite.Filter.class})
+public class TestAuthzBeforeRewrite
```

Comment:
wdyt of this? 

```suggestion
public class TestSecurityControllerCalledBeforeRewrite
```

`Authorization` is a loaded term because its also the name of a SigV4 header

we usually use `Security` to denote stuff that is for access control

i would also move this into the `io.trino.aws.proxy.server.security` package 

sorry to nit-pick

Reply:
I will rename it, however my understanding of the current structure is that package is for used for tests related to the controller and OPA only, which is why I suggest we don't move it there. Unless I'm mistaken in which case happy to move it there.